### DISCUSSION TITLE
Composing attribute from nested values

### DISCUSSION THREAD

**Discussion by junaway** [sensedeep/dynamodb-onetable#139] [General]
Hi, what is the right way to compose a string attribute from another nested attribute value ?

    User: {
        pk:      { type: String, value: '${data.id}' },
        data:    { type: Object, schema: {
            id:  { type: String }
        }
    }

Something like this but not this.

**Comment by mobsense** [comment]
We don't (yet) support nested properties in value templates.

You need to hoist the ID to be a top level property.

Can you please open an issue for this or let me know and I'll create one for it.

**Comment by junaway** [comment]
Hi mobsense,

Here you go: [Issue](https://github.com/sensedeep/dynamodb-onetable/issues/140)

**Comment by mobsense** [comment]
Should now be fully fixed in 1.9.2 prerelease.
